shop/src/components/payment/crypto/CryptoDirect.js

```diff
@@ -92,22 +95,44 @@ const PayWithCryptoDirect = ({ submit, encryptedData, onChange, loading }) => {
 
   const acceptedTokens = config.acceptedTokens
 
+  useEffect(() => {
+    async function queryWalletBalance() {
+      return await wallet.signer.getBalance()
+    } //returns a promise that, if resolved, will give the balance of the wallet in wei (BigNumber)
```

Comment:
If you only need this function to return promise, you remove the `async` and `await` to keep code simple. It'll still return a promise that you can `await`